storage/innobase/row/row0mysql.cc

```diff
@@ -2135,6 +2208,14 @@ row_update_for_mysql(row_prebuilt_t* prebuilt)
 		prebuilt->table->stat_modified_counter++;
 	}
 
+	if (DICT_TF2_FLAG_IS_SET(node->table, DICT_TF2_VERSIONED) &&
+		(node->versioned || node->vers_delete ||
+			// TODO: imrove this check (check if we touch only
+			// unversioned fields in foreigh table
+			node->foreign)) {
```

Comment:
Could a member function be added to `node`? Something like `if (node->some_descriptive_predicate_name()) trx->vtq_notify_on_commit = true;`